src/background_thread.c

```diff
@@ -11,12 +11,14 @@
 #define BACKGROUND_THREAD_DEFAULT false
 /* Read-only after initialization. */
 bool opt_background_thread = BACKGROUND_THREAD_DEFAULT;
+size_t opt_max_background_threads = SIZE_T_MAX;
```

Comment:
nit: add a `#define MAX_BACKGROUND_THREAD_LIMIT MALLOCX_ARENA_LIMIT`? It's probably safe to assume we will never have more than 1 thread per arena.